🐛 BUG: Transform couldn't find `astro/internal`

### What version of `@astrojs/compiler` are you using?

0.14.2

### What package manager are you using?

npm

### What operating system are you using?

Windows

### Describe the Bug

When using the `transform` method imported from @astrojs/compiler, I get this error, saying that `astro/internal` couldn't be found:

```
Cannot find module 'astro/internal' imported from '/home/projects/github-dm6xxp-hnfnvq/src/pages/index.astro'
Error: Cannot find module 'astro/internal' imported from '/home/projects/github-dm6xxp-hnfnvq/src/pages/index.astro'
    at viteResolve (/home/projects/github-dm6xxp-hnfnvq/node_modules/vite/dist/node/chunks/dep-59dc6e00.js:59357:25)
    at nodeImport (/home/projects/github-dm6xxp-hnfnvq/node_modules/vite/dist/node/chunks/dep-59dc6e00.js:59392:15)
    at ssrImport (/home/projects/github-dm6xxp-hnfnvq/node_modules/vite/dist/node/chunks/dep-59dc6e00.js:59288:20)
    at eval (/home/projects/github-&#8230;
```

Note:
* Changing `internalURL` to the absolute path makes a new error appear:
```
failed to load module for ssr: ../astro/internal
Error: failed to load module for ssr: ../astro/internal
    at instantiateModule (/home/projects/github-dm6xxp-hnfnvq/node_modules/vite/dist/node/chunks/dep-59dc6e00.js:59253:15)
``` 

I tried adding `astro` to `vite.ssr.external` and `vite.optimizeDeps.exclude` (via updateConfig or as part of the original `astro.config.mjs` file) but the issue persists.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-dm6xxp-hnfnvq?file=packages%2Fplugin.js,astro.config.mjs

--- a/.changeset/short-crabs-chew.md
+++ b/.changeset/short-crabs-chew.md
@@ -1,0 +1,5 @@
+---
+'@astrojs/compiler': patch
+---
+
+Update default `internalURL` to `astro/runtime/server/index.js`

--- a/cmd/astro-wasm/astro-wasm.go
+++ b/cmd/astro-wasm/astro-wasm.go
@@ -86,7 +86,7 @@ func makeTransformOptions(options js.Value) transform.TransformOptions {
 
 	internalURL := jsString(options.Get("internalURL"))
 	if internalURL == "" {
-		internalURL = "astro/internal"
+		internalURL = "astro/runtime/server/index.js"
 	}
 
 	sourcemap := jsString(options.Get("sourcemap"))

--- a/packages/compiler/README.md
+++ b/packages/compiler/README.md
@@ -17,7 +17,7 @@ The Astro compiler can convert `.astro` syntax to a TypeScript Module whose defa
 **Some notes**...
 
 - TypeScript is valid `.astro` syntax! The output code may need an additional post-processing step to generate valid JavaScript.
-- `.astro` files rely on a server implementation exposed as `astro/internal` in the Node ecosystem. Other runtimes currently need to bring their own rendering implementation and reference it via `internalURL`. This is a pain point we're looking into fixing.
+- `.astro` files rely on a server implementation exposed as `astro/runtime/server/index.js` in the Node ecosystem. Other runtimes currently need to bring their own rendering implementation and reference it via `internalURL`. This is a pain point we're looking into fixing.
 
 ```js
 import { transform } from '@astrojs/compiler';
@@ -26,7 +26,7 @@ const result = await transform(source, {
   site: 'https://mysite.dev',
   sourcefile: '/Users/astro/Code/project/src/pages/index.astro',
   sourcemap: 'both',
-  internalURL: 'astro/internal',
+  internalURL: 'astro/runtime/server/index.js',
 });
 ```
 

